src/h3lib/lib/localij.c

```diff
@@ -30,6 +30,83 @@
 #include "mathExtensions.h"
 #include "stackAlloc.h"
 
+/**
+ * Origin leading digit -> index leading digit -> rotations 60 cw
+ * Either being 1 (K axis) is invalid.
+ * No good default at 0.
+ */
+const int PENTAGON_ROTATIONS[7][7] = {
+    {0, -1, 0, 0, 0, 0, 0},        // 0
+    {-1, -1, -1, -1, -1, -1, -1},  // 1
+    {0, -1, 0, 0, 0, 1, 0},        // 2
+    {0, -1, 0, 0, 1, 1, 0},        // 3
+    {0, -1, 0, 5, 0, 0, 0},        // 4
+    {0, -1, 5, 5, 0, 0, 0},        // 5
+    {0, -1, 0, 0, 0, 0, 0},        // 6
+};
+/**
+ * Reverse base cell direction -> leading index digit -> roations 60 ccw.
+ * For reversing the rotation introduced in PENTAGON_ROTATIONS when
+ * the origin is on a pentagon.
+ */
+const int PENTAGON_ROTATIONS_REVERSE[7][7] = {
```

Comment:
I get `PENTAGON_ROTATIONS`, `PENTAGON_ROTATIONS_REVERSE_NONPOLAR` and `PENTAGON_ROTATIONS_REVERSE_POLAR`, but I don't get this one. Is this one applied in both cases, but then why have the other two? Or why not simply push the compound rotation results into both of the ones below if this is applied and then one of the following two are applied?

Reply:
`PENTAGON_ROTATIONS_REVERSE` is not used in the same cases. The comments try to differentiate that this table is used when the origin index is on a pentagon base cell, the two "reverse" tables are used when the origin index is not on a pentagon base cell.